infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/PodTolerationProvisionerTest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.provision;
+
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.testng.Assert.assertEquals;
+
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.workspace.infrastructure.kubernetes.environment.KubernetesEnvironment;
+import org.eclipse.che.workspace.infrastructure.kubernetes.environment.KubernetesEnvironment.PodData;
+import org.mockito.Mock;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+/**
+ * Tests {@link PodTolerationProvisioner}.
+ *
+ * @author Masak Muranaka
+ */
+@Listeners(MockitoTestNGListener.class)
+public class PodTolerationProvisionerTest {
+
+  @Mock private KubernetesEnvironment k8sEnv;
+  @Mock private RuntimeIdentity runtimeIdentity;
+
+  @Test
+  public void doNothingWhenSttringIsNull() throws Exception {
+    // given
+    PodTolerationProvisioner podTolerationProvisioner = new PodTolerationProvisioner(null);
+
+    // when
+    podTolerationProvisioner.provision(k8sEnv, runtimeIdentity);
+
+    // then
+    verify(k8sEnv, never()).getIngresses();
+  }
+
+  @Test
+  public void doNothingWhenSttringIsEmpty() throws Exception {
+    // given
+    PodTolerationProvisioner podTolerationProvisioner = new PodTolerationProvisioner("");
+
+    // when
+    podTolerationProvisioner.provision(k8sEnv, runtimeIdentity);
+
+    // then
+    verify(k8sEnv, never()).getIngresses();
+  }
+
+  @Test
+  public void justOnlyEffectIsSpecified() throws Exception {
```

Comment:
We should include a test for `<effect>:<key>::<tolerationSeconds>:<value>` to verify that `tolerationSeconds` and `value` are set.